### PR TITLE
Add control to disable / bypass CRL download failures

### PR DESCRIPTION
**Pull request by rhurey**
Add the ability to skip CRL checks, or continue on CRL download failure.

### CODE REVIEW

File: adapters/tlsio_openssl.c [hunk 1]
```diff
@@ -1607,6 +1619,21 @@ static STACK_OF(X509_CRL) *crls_http_cb(X509_STORE_CTX *ctx, X509_NAME *nm)
...
+{
+    if(X509_V_ERR_UNABLE_TO_GET_CRL == X509_STORE_CTX_get_error(ctx))
+    {
+        LogInfo("Ignoring CRL Downlaod failure\n");
+        return 1;
+    }
+    else
+    {
+        LogError("Error %d was unexpected\n", X509_STORE_CTX_get_error(ctx));
+        return status;
```
Comment (ralph-msft) [inline adapters/tlsio_openssl.c:1632]: Could the value of ```status``` also be set to 1? #Resolved

File: adapters/tlsio_openssl.c [hunk 2]
```diff
@@ -2238,6 +2271,7 @@ CONCRETE_IO_HANDLE tlsio_openssl_create(void* io_create_parameters)
 
                     result->tls_version = OPTION_TLS_VERSION_1_0;
                     result->disable_crl_check = false;
+                    result->continue_on_crl_download_failure = false;
                     result->disable_default_verify_paths = false;
```
Comment (ralph-msft) [inline adapters/tlsio_openssl.c:2275]: Nitpick: Would it be easier to do instead do ```memset(&result, 0, sizeof(result))``` here, and then set any non-zero defaults (e.g. for tls_version)?

File: inc/azure_c_shared_utility/shared_util_options.h [hunk 3]
```diff
@@ -25,6 +25,7 @@ extern "C"
     static STATIC_VAR_UNUSED const char* const OPTION_TRUSTED_CERT = "TrustedCerts";
 
     static STATIC_VAR_UNUSED const char* const OPTION_DISABLE_CRL_CHECK = "DisableCrlCheck";
+    static STATIC_VAR_UNUSED const char* const OPTION_CONTINUE_ON_CRL_DOWNLOAD_FAILURE = "AllowCRLFetchFailure";
```
Comment (ralph-msft) [inline inc/azure_c_shared_utility/shared_util_options.h]: Nitpick: The value you've used here value doesn't seem to be consistent with the other values used. Have you considered ```ContinueOnCrlDownloadFailure``` instead? #Resolved

**Review by ralph-msft — APPROVED** [review]

File: inc/azure_c_shared_utility/shared_util_options.h (hunk 3 above)
Comment (rhurey) [reply inc/azure_c_shared_utility/shared_util_options.h]: Doh! I renamed it, and my find / replace missed that one.

Thanks.

File: adapters/tlsio_openssl.c (hunk 1 above)
Comment (rhurey) [reply adapters/tlsio_openssl.c:1632]: Could.

From the OpenSSL Docs:

> The ok parameter to the callback indicates the value the callback should return to retain the default behaviour. If it is zero then an error condition is indicated. If it is 1 then no error occurred.

https://www.openssl.org/docs/man1.1.0/man3/X509_STORE_CTX_set_verify_cb.html

I only want to change the behavior for the CRL get operation, so I return the default behavior code for the rest.

Comment (ralph-msft) [reply adapters/tlsio_openssl.c:1632]: Got it.

I also missed this earlier but you have typo at line 1626: ```Downlaod``` instead of ```Download```

**Pull request merged**